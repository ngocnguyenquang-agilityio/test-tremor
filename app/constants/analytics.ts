/* Constants for Analytics Page */
export const ANALYTICS_STATISTICAL_TYPE = {
  BOOKINGS: "Bookings",
  REVENUE: "Revenue",
  TODAY_USER: "Today's Users",
  FOLLOWERS: "Followers",
};

export const enum CHART_TYPE {
  SALE = "sale",
  PERFORMANCE = "performance",
}
