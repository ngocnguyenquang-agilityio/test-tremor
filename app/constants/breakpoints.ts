export const BREAKPOINTS = {
  MD: 768,
  LG: 1024,
};
