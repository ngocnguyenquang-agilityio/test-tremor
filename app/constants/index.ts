export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./sections";
export * from "./images";
