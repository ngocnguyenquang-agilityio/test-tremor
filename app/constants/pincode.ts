const DEFAULT_PIN_CODE = "*";
const PIN_CODE_LENGTH = 6;

export { DEFAULT_PIN_CODE, PIN_CODE_LENGTH };
