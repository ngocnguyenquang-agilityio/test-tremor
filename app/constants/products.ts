// Types
import { SelectOptionData } from "@/types";

type TopSellingProductsType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
  5: string;
};

export const TOP_SELLING_PRODUCTS_SRC: TopSellingProductsType = {
  1: "/assets/images/products/blue-shoe.webp",
  2: "/assets/images/products/black-mug.webp",
  3: "/assets/images/products/black-chair.webp",
  4: "/assets/images/products/bang-sound.webp",
  5: "/assets/images/products/photo-tools.webp",
};

export const TYPE_PRICE: SelectOptionData[] = [
  {
    option: "BTC",
    value: "1",
  },
  {
    option: "CNY",
    value: "2",
  },
  {
    option: "EUR",
    value: "3",
  },
  {
    option: "GBP",
    value: "4",
  },
  {
    option: "INR",
    value: "5",
  },
  {
    option: "USD",
    value: "6",
  },
];

export const TAGS_PRICE: SelectOptionData[] = [
  {
    option: "Black Friday",
    value: "1",
  },
  {
    option: "Expired",
    value: "2",
  },
  {
    option: "Out of Stock",
    value: "3",
  },
  {
    option: "In Stock",
    value: "4",
  },
  {
    option: "Sale",
    value: "5",
  },
];

export const CATEGORY_PRODUCT: SelectOptionData[] = [
  {
    option: "Clothing",
    value: "1",
  },
  {
    option: "Electronics",
    value: "2",
  },
  {
    option: "Furniture",
    value: "3",
  },
  {
    option: "Others",
    value: "4",
  },
  {
    option: "Real Estate",
    value: "5",
  },
];

export const SIZE_PRODUCT: SelectOptionData[] = [
  {
    option: "Extra Large",
    value: "1",
  },
  {
    option: "Extra Small",
    value: "2",
  },
  {
    option: "Large",
    value: "3",
  },
  {
    option: "Medium",
    value: "4",
  },
  {
    option: "Small",
    value: "5",
  },
];
