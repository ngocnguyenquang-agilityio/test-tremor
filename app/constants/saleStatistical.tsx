import { ReactNode } from "react";
import { IoMdPersonAdd } from "react-icons/io";
import { MdLeaderboard, MdStore, MdWeekend } from "react-icons/md";

interface Icon {
  bgIcon: string;
  icon: ReactNode;
}

interface Sale {
  [key: string]: Icon;
}

export const SALE_STATISTICAL: Sale = {
  bookings: {
    bgIcon: "bg-gradient-arsenic",
    icon: <MdWeekend color="white" size="24px" />,
  },
  todayUser: {
    bgIcon: "bg-[linear-gradient(195deg,#49a3f1,#1A73E8)]",
    icon: <MdLeaderboard color="white" size="24px" />,
  },
  revenue: {
    bgIcon: "bg-[linear-gradient(195deg,#66BB6A,#43A047)]",
    icon: <MdStore color="white" size="24px" />,
  },
  followers: {
    bgIcon: "bg-[linear-gradient(195deg,#EC407A,#D81B60)]",
    icon: <IoMdPersonAdd color="white" size="24px" />,
  },
};
