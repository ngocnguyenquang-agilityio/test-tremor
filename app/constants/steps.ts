import { Step } from "@/types";

export const NEW_PRODUCT_FORM_STEPS: Step[] = [
  { index: 1, content: "Product Info" },
  { index: 2, content: "Media" },
  { index: 3, content: "Social" },
  { index: 4, content: "Pricing" },
];
