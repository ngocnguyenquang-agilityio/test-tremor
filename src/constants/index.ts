export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
