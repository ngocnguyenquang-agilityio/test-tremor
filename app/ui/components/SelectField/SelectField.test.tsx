import { render, fireEvent } from "@testing-library/react";

// Constants
import { TYPE_PRICE } from "@/constants";

// Components
import { SelectField } from "..";

describe("SelectField Component", () => {
  it("Should match snapshot", () => {
    const component = render(
      <SelectField label="Currency" options={TYPE_PRICE} />,
    );
    expect(component).toMatchSnapshot();
  });

  it("renders correct label text", () => {
    const { getByText } = render(
      <SelectField label="Currency" options={TYPE_PRICE} />,
    );

    const label = getByText("Currency");

    expect(label).toBeTruthy;
  });

  it("renders correct number of options", () => {
    const { getByTestId } = render(
      <SelectField
        label="Currency"
        options={TYPE_PRICE}
        dataTestId="currency"
      />,
    );
    const selectElement = getByTestId("select-field");
    expect(selectElement.children).toHaveLength(TYPE_PRICE.length);
  });

  it("renders correct name of options", () => {
    const { getByText } = render(
      <SelectField
        label="Currency"
        options={TYPE_PRICE}
        dataTestId="currency"
      />,
    );
    expect(getByText("GBP")).toBeTruthy();
  });
});
