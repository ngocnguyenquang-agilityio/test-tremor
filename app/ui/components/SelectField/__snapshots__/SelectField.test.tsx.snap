// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full"
      >
        <p
          class="text-tremor-default text-secondary text-sm dark:text-lighter"
        >
          Currency
        </p>
        <select
          class="peer h-full w-full appearance-none rounded-none border-b-2 border-gray-300 bg-transparent font-sans text-sm font-normal text-primary dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer outline outline-0 transition-all placeholder-shown:border-blue-gray-200 focus:border-gray-500 focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50 placeholder:opacity-0 focus:placeholder:opacity-100 undefined"
          data-testid="select-field"
        >
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="1"
          >
            BTC
          </option>
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="2"
          >
            CNY
          </option>
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="3"
          >
            EUR
          </option>
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="4"
          >
            GBP
          </option>
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="5"
          >
            INR
          </option>
          <option
            class="dark:bg-primary"
            data-testid="select-option"
            value="6"
          >
            USD
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full"
    >
      <p
        class="text-tremor-default text-secondary text-sm dark:text-lighter"
      >
        Currency
      </p>
      <select
        class="peer h-full w-full appearance-none rounded-none border-b-2 border-gray-300 bg-transparent font-sans text-sm font-normal text-primary dark:text-white dark:border-gray-600 dark:focus:border-gray-500 focus:outline-none focus:ring-0 focus:border-gray-600 peer outline outline-0 transition-all placeholder-shown:border-blue-gray-200 focus:border-gray-500 focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50 placeholder:opacity-0 focus:placeholder:opacity-100 undefined"
        data-testid="select-field"
      >
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="1"
        >
          BTC
        </option>
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="2"
        >
          CNY
        </option>
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="3"
        >
          EUR
        </option>
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="4"
        >
          GBP
        </option>
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="5"
        >
          INR
        </option>
        <option
          class="dark:bg-primary"
          data-testid="select-option"
          value="6"
        >
          USD
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
